stock_inventory/tests/test_stock_inventory.py

```diff
@@ -555,3 +555,29 @@ def test_12_search_products_under_review(self):
             expected_result,
             "The search function did not return the expected results",
         )
+
+    def test_13_multiple_inventories_different_products_same_location(self):
+        inventory1 = self.inventory_model.create(
+            {
+                "name": "Inventory1 for Product1",
+                "product_ids": [(6, 0, [self.product.id])],
+                "location_ids": [(6, 0, [self.location3.id])],
+                "product_selection": "manual",
+            }
+        )
+        inventory2 = self.inventory_model.create(
+            {
+                "name": "Inventory2 for Product2",
+                "product_ids": [(6, 0, [self.product2.id])],
+                "location_ids": [(6, 0, [self.location3.id])],
+                "product_selection": "manual",
+            }
+        )
+        inventory1.action_state_to_in_progress()
+        inventory2.action_state_to_in_progress()
+        self.assertEqual(inventory1.state, "in_progress")
+        self.assertEqual(inventory2.state, "in_progress")
+        self.quant3.inventory_quantity = 80
+        self.quant3.action_apply_inventory()
```

Comment:
@JoanSForgeFlow Shouldn't be the test call `_action_apply_inventory`on the recordset composed by quant 3 and 4 ?